### ISSUE TITLE
Thread warning when adding destination

### ISSUE THREAD

**Issue by mickyzinho** [braze-inc/braze-segment-swift#11]
When adding the BrazeDestination to Segment via `add(plugin: brazeDestination)` and setting `configuration.push.automation = true`, I always get the warning:
```
Unable to perform push notification automation steps. The SDK will not be able to automatically process push notifications.
- reason: The SDK must be initialized on the main thread to enable push automation.
```
Wrapping the call in main dispatch or using MainActor does not change anything. Does the integration always dispatch the Braze initialisation to a background queue?

**Comment by mickyzinho** [comment]
Hi @lowip ,
thank you for the quick response and the workaround! 
We'll go with `1.4.8` then in the meantime.

**Comment by robcreatively** [comment]
Hi @lowip ,

Any update here? Also encountering the issue and analytics-swift is up to version 1.5.6 now.

Thanks!

**Comment by samsiner** [comment]
Hi @lowip - is there any update yet?  We're also encountering this issue, and analytics-swift is up to version 1.5.8 as of today, so we don't want to pin it too far behind. Thanks!

**Comment by lowip** [comment]
Hi @mickyzinho, @robcreatively and @samsiner,

We are actively working on this issue and should be able to provide meaningful updates soon. Resolving this issue requires important changes to the way our Swift SDK handles push notifications. The impact of such change is not limited to our Segment integration, and thus, requires extensive testing.

Thank you for your patience!

**Comment by carlos-prezzee** [comment]
Is there any update on this? Our spent a lot of time debugging and issue with Braze not being initialised on time due to this problem

**Comment by mickyzinho** [comment]
@lowip Any news on the schedule for fixing this? Pinning the `analytics-swift` dependency leaves us with being unable to submit any updates, as Apple now requires the privacy information changes which have been added with `1.5.8`. This is quite urgent for us!

**Comment by lowip** [comment]
Hi everyone,

We're planning to release a fix for this issue in the next couple of days. Stay tuned, we'll update this thread ASAP when the fix will be released!

**Comment by lowip** [comment]
Hey everyone,

Thank you for your patience, we have released version [`3.0.0`](https://github.com/braze-inc/braze-segment-swift/releases/tag/3.0.0) which allows you to use versions `1.5.0+` of https://github.com/segmentio/analytics-swift and keep the push notification support working.

Please be aware that this is a major release which requires a small change in the integration. All details can be found in the [changelog entry](https://github.com/braze-inc/braze-segment-swift/blob/main/CHANGELOG.md#300).
